File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function(nums1, nums2) {
     const set1 = new Set(nums1);
     const set2 = new Set(nums2);

    return [[...set1].filter(num => !set2.has(num)), [...set2].filter(num => !set1.has(num))];
};

File: 2215-find-the-difference-of-two-arrays/2215-find-the-difference-of-two-arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        vector<vector<int>> Diff(2);
        unordered_set<int> set1(nums1.begin(), nums1.end());
        unordered_set<int> set2(nums2.begin(), nums2.end());
        for(auto num : set1)
            if (!set2.count(num))
                Diff[0].push_back(num);
        for(auto num : set2)
            if (!set1.count(num))
                Diff[1].push_back(num);
        return Diff;
        
    }
};